identity_stardust/src/state_metadata/document.rs

```diff
@@ -35,21 +36,22 @@ pub(crate) struct StateMetadataDocument {
 
 impl StateMetadataDocument {
   /// Transforms the document into a [`StardustDocument`] by replacing all placeholders with `original_did`.
-  pub fn into_stardust_document(self, original_did: &CoreDID) -> StardustDocument {
+  pub fn into_stardust_document(self, original_did: &StardustDID) -> Result<StardustDocument> {
     let Self { document, metadata } = self;
-    let core_document: StardustCoreDocument = document.map(
+    let core_document: StardustCoreDocument = document.try_map(
       // Replace placeholder identifiers.
       |did| {
         if did == PLACEHOLDER_DID.as_ref() {
-          original_did.clone()
+          Ok(original_did.clone())
         } else {
-          did
+          // TODO: wrap error?
```

Comment:
If there was no internal use of this method it would be fine to leave it as is as then the context is clear to the caller. Currently we call this method in `StardustDocument::unpack`, but I am not sure to which extent more context is useful for the caller in that case.